extensions/jdbc/jdbc-postgresql/deployment/src/test/java/io/quarkus/jdbc/postgresql/deployment/DevServicesPostgresqlDatasourceWithVolumeTestCase.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.jdbc.postgresql.deployment;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.ResultSet;
+
+import javax.sql.DataSource;
+
+import jakarta.inject.Inject;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class DevServicesPostgresqlDatasourceWithVolumeTestCase {
```

Comment:
Note you could probably take advantage of `QuarkusDevModeTest` to test that data is persisted upon restart... Though I don't know how to force a restart, except by modifying a file like in `io.quarkus.hibernate.orm.HibernateHotReloadTestCase#testImportSqlWithContinuousTesting`.

Reply:
This test is not mapping the db data, just using a volume where there is a script. 
Maybe we could add another test case to verify this use case using QuarkusDevModeTest (actually, I have these tests in another repository because I wanted to be sure about it), but I don't think it's really necessary to be covered here.

Reply:
The container is not restarted on hot reload, so testing by modifying the file does not really prove anything with regards to how container volumes are handled.

Reply:
Right. We'd need to modify `application.properties` and change very specific configuration properties to trigger a restart:

https://github.com/quarkusio/quarkus/blob/5e78877a5f0c6592e9663824b8536a4f44288904/extensions/datasource/deployment/src/main/java/io/quarkus/datasource/deployment/devservices/DevServicesDatasourceProcessor.java#L75-L101

Reply:
I think this is out of the scope of this pull request.

Reply:
I think testing the features we document in a PR ("is handy [...] to preserve database data and reuse it after an application restart") is very much in scope of that PR.
But I'll grant you this opinion isn't widely shared.